enterprise/reporting/setup_test.go

```diff
@@ -77,8 +82,8 @@ func TestSetupForNoop(t *testing.T) {
 					EnterpriseToken: "dummy-token",
 				}
 			}
-			med := NewReportingMediator(logger.NOP, f.EnterpriseToken)
-			require.Equal(t, med.createErrorReportInstance(&backendconfig.NOOP{}), &NOOP{})
+			med := NewReportingMediator(context.Background(), logger.NOP, f.EnterpriseToken, &backendconfig.NOOP{})
+			require.Len(t, med.delegates, tc.expectedDelegates)
```

Comment:
[non-blocking]
I prefer a more black-box approach when it comes to testing here. 

* No accessing unexported fields like `instance` and `delegates`
* Not relying on an implementation detail of a delegate slice to tests that delegates are enabled
* Using `reporting_test` package instead of `reporting` 

I understand this is an incremental step, but I just wanted to express my thoughts when looking at this code.